File: src/js/sort_builtin.js
var fs = require("fs");

//////////////////////////
// business logic start //
//////////////////////////

function doSorting(inputList) {
  outputList = inputList;
  outputList.sort();
  return outputList;
}

////////////////////////
// business logic end //
////////////////////////

// 👇🏽 copy pasted helpers

// read input file
let inputFilePath = process.env.INPUT_PATH;
let inputString = fs.readFileSync(inputFilePath, "utf8");
let inputStringSplit = inputString.split(/\n/);

// clean input data
if (inputStringSplit[inputStringSplit.length - 1] === "") {
  inputStringSplit = inputStringSplit.slice(0, inputStringSplit.length - 1);
}

// do sorting
sortedData = doSorting(inputStringSplit);

// write output file
let outputFilePath = process.env.OUTPUT_PATH;
let outputString = sortedData.join("\n");
outputString += "\n"; // trailing newline
fs.writeFileSync(outputFilePath, outputString);

File: src/ruby/sort_builtin.rb
# frozen_string_literal: true

########################
# business logic start #
########################

def do_sorting(input_list)
  output_list = input_list
  output_list = output_list.sort
  output_list
end

######################
# business logic end #
######################

if __FILE__ == $PROGRAM_NAME
  input_file_path = ENV['INPUT_PATH']
  input_file = File.readlines(input_file_path)

  sorted_data = do_sorting(input_file)

  output_file_path = ENV['OUTPUT_PATH']
  File.open(output_file_path, 'wb') { |f| f.write(sorted_data.join('')) }
end
